libs/datamodel/connectors/sql-datamodel-connector/src/mysql_datamodel_connector.rs

```diff
@@ -156,45 +156,64 @@ impl Connector for MySqlDatamodelConnector {
     }
 
     fn validate_field(&self, field: &Field) -> Result<(), ConnectorError> {
-        if let FieldType::NativeType(_, native_type) = field.field_type() {
-            let native_type_name = native_type.name.as_str();
-            if matches!(native_type_name, DECIMAL_TYPE_NAME | NUMERIC_TYPE_NAME) {
-                match native_type.args.as_slice() {
-                    [precision, scale] if scale > precision => {
-                        return Err(ConnectorError::new_scale_larger_than_precision_error(
-                            native_type_name,
-                            "MySQL",
-                        ))
-                    }
-                    [precision, _] if *precision > 65 => {
-                        return Err(ConnectorError::new_argument_m_out_of_range_error(
-                            "Precision can range from 1 to 65.",
-                            native_type_name,
-                            "MySQL",
-                        ))
-                    }
-                    [_, scale] if *scale > 30 => {
-                        return Err(ConnectorError::new_argument_m_out_of_range_error(
-                            "Scale can range from 0 to 30.",
-                            native_type_name,
-                            "MySQL",
-                        ))
-                    }
-                    _ => {}
+        if let FieldType::NativeType(_, native_type_instance) = field.field_type() {
+            let native_type_name = native_type_instance.name.as_str();
+            let native_type: MySqlType = native_type_instance.deserialize_native_type();
+
+            let precision_and_scale = match native_type {
+                MySqlType::Decimal(x) => x,
+                MySqlType::Numeric(x) => x,
```

Comment:
tiny nitpick (no need to change), but here `MySqlType::Decimal(x) | MySqlType::Numeric(x) => x` would work as well.